### ISSUE TITLE
[1300.0.8] Weapon Bots dont shoot ice spires

### ISSUE THREAD

**Issue by CurePindal** [Regalis11/Barotrauma#5504]
- [x] I have searched the issue tracker to check if the issue has already been reported.

**Description**
When a player drives the sub bots on weapons shoot ice spires on their own. (not ice chunks)
When a bot drives the sub it usually stops when it spots an ice spire (safety first) but bot on the guns suddenly dont shoot spires at all.

Expected: Same behaviour as if a player drives the sub: bots still aim and shoot ice spires to clear the way.

**Steps To Reproduce**
Described above

**Version**
[1300.0.8]  win10, sp, vanilla

**Comment by itchyOwl** [comment]
It doesn't matter who drives the sub, it's just because the sub doesn't move. The bots shoot the spires only when they are really close or when the sub is moving towards them. The intention for this was that the player might not want the bots to waste bullets on on spires that are not on the way.

But I agree that there might be an issue here: If the sub should be moving towards a spire that blocks the path, the auto steering stops, which in turn makes the bots not shoot the spire that's blocking that path. So I think we'd actually need to introduce a different condition for the sub that is navigated by a bot.

Need to think about it.

**Comment by CurePindal** [comment]
Yes, its indeed an issue that the helm bot stops at almost every spire/spike and drops a radio message thats faster faded out than noticed.

I would love to see 2 things
1. Helm bot moves around spires if possible
2. Helm bot can move a manned gun (bot or player) into LOS to shoot the spire.

In general I would love to see more indeep access to the state machine of the bots to set priorities for tasks. 
For reference, Pulsar: Lost Colony did a pretty decent job to adjust the bot state machine ingame either quickNdirty or indeep and complex.

**Comment by itchyOwl** [comment]
We are planning to add a feature where you can see the objective the bot is doing, when it's not following the order for some reason. But I don't think we are going to add micromanaging that AI. More control over it maybe.

Btw, bot and npc AI (=the same) is using behavior tree, not FSM, unlike the monsters.

It was a deliberate decision to not make the auto steering go around the spires. I think @Regalis11 can answer to that better than I.

**Comment by Regalis11** [comment]
Yeah, the reason we made the AI very (or perhaps even too) cautious around spires was that if the AI is good enough to navigate past them, they're essentially an obstacle you don't need to care about at all, but if there's a chance the AI messes up and hits them, it could lead to frustrating situations and encourage the players to never rely on the bots/autopilot to steer the sub. 

In earlier unstable versions the AI did sometimes hit the spires or the level walls when trying to dodge a spire, and we got a lot of negative feedback about it (here's on example https://github.com/Regalis11/Barotrauma/issues/4666).

So I'd personally leave towards keeping the current behavior, or at least making sure the AI never accidentally steers into the spires.
